tests/unit/__init__.py

```diff
@@ -0,0 +1,89 @@
+#See the file COPYING for more details.
+#Copyright (C) 2017 NV Access Limited
+
+"""NVDA unit testing.
+All unit tests should reside within this package and should be
+divided into modules and packages similar to the code they are testing.
+Test modules must have a C{test_} prefix
+and should contain one or more classes with a C{Test} prefix which subclass L{TestCase}.
+Methods in test classes should have a C{test_} prefix.
+"""
+
+### Ugly bootstrapping code.
+
+import os
+import sys
+
+# The path to the unit tests.
+UNIT_DIR = os.path.dirname(os.path.abspath(__file__))
+# The path to the top of the repo.
+TOP_DIR = os.path.dirname(os.path.dirname(UNIT_DIR))
+# The path to the NVDA "source" directory.
+SOURCE_DIR = os.path.join(TOP_DIR, "source")
+# Let us import modules from the NVDA source.
+sys.path.insert(1, SOURCE_DIR)
+import sourceEnv
+
+# Set options normally taken from the command line.
+import globalVars
+class AppArgs:
+	# The path from which to load a configuration file.
+	# Ideally, this would be an in-memory, default configuration.
+	# However, config currently requires a path.
+	# We use the unit test directory, since we want a clean config.
+	configPath = UNIT_DIR.decode("mbcs")
+	secure = False
+	disableAddons = True
+globalVars.appArgs = AppArgs()
+
+# We depend on the current directory to load some files;
+# e.g. braille imports louis which loads liblouis.dll using a relative path.
+os.chdir(SOURCE_DIR)
+# The path to this package might be relative, so make it absolute,
+# since we just changed directory.
+__path__[0] = UNIT_DIR
+# We don't want logging.
+import logging
+from logHandler import log
+log.addHandler(logging.NullHandler())
+# There's no point in logging anything at all, since it'll go nowhere.
+log.setLevel(100)
+
+# Much of this should eventually be replaced by stuff which gets reset before each test
+# so the tests are isolated.
+import config
+config.initialize()
+# Initialize languageHandler so that translatable strings work.
+import languageHandler
+languageHandler.setLanguage("en")
+# NVDAObjects need appModuleHandler to be initialized.
+import appModuleHandler
+appModuleHandler.initialize()
+# Anything which notifies of cursor updates requires braille to be initialized.
+import braille
+braille.initialize()
+# The focus and navigator objects need to be initialized to something.
+from NVDAObjects import NVDAObject
+class PlaceholderNVDAObject(NVDAObject):
+	processID = None # Must be implemented to instantiate.
+phObj = PlaceholderNVDAObject()
+import api
+api.setFocusObject(phObj)
+api.setNavigatorObject(phObj)
+
+# Stub speech functions to make them no-ops.
+# Eventually, these should keep track of calls so we can make assertions.
+import speech
+speech.speak = lambda speechSequence, symbolLevel=None: None
+speech.speakSpelling = lambda text, locale=None, useCharacterDescriptions=False: None
+
+### Base classes/utility functions.
+
+import unittest
+
+class TestCase(unittest.TestCase):

```

Comment:
As discussed, I dont think we need this for now. We will need some way of managing setup/tear down of dependencies, but perhaps we should address this later.